lib/gui/components/drive-selector/controllers/drive-selector.js

```diff
@@ -170,4 +172,81 @@ module.exports = function(
     });
   };
 
+  /**
+   * @summary Memoize ImmutableJS list reference
+   * @function
+   * @private
+   *
+   * @description
+   * This workaround is needed to avoid AngularJS from getting
+   * caught in an infinite digest loop when using `ngRepeat`
+   * over a function that returns a mutable version of an
+   * ImmutableJS object.
+   *
+   * The problem is that every time you call `myImmutableObject.toJS()`
+   * you will get a new object, whose reference is different from
+   * the one you previously got, even if the data is exactly the same.
+   *
+   * @param {Function} func - function that returns an ImmutableJS list
+   * @param {Function} compare - function used to compare the old and new lists
+   * @returns {Function} memoized function
+   *
+   * @example
+   * const getList = () => {
+   *   return Store.getState().toJS().myList;
+   * };
+   *
+   * const memoizedFunction = memoizeImmutableListReference(getList);
+   */
+  const memoizeImmutableListReference = (func) => {
```

Comment:
I don't think we should bother about that. This "hack" is very focused, and will get removed once we start moving stuff to react.